src/pkcs1_5.c

```diff
@@ -607,22 +588,21 @@ int pkcs_v1_5_sign_man_ace_v1_5(struct image *image,
 	}
 
 	/* calculate the digest - SHA384 on CAVS2_5+ */
-	module_sha384_create(image);
-	module_sha_update(image, ptr1, size1);
-	module_sha_update(image, ptr2, size2);
-	module_sha_complete(image, digest);
+	hash_sha384_init(&digest);
+	hash_update(&digest, ptr1, size1);
+	hash_update(&digest, ptr2, size2);
```

Comment:
check ec... and for all next occurencies :)

Reply:
The error code is stored in hash_context. You only need to check it once at the end

Reply:
So if you call 3 functions in a row and then check error code, do you know which function failed? Error reporting becomes less readable and precise.
Is it always allowed to call next function if previous one failed?